### PR TITLE
Including CI configs and pre-commit fix

### PR DESCRIPTION
**Pull request by harshad16**
Including CI configs and pre-commit fix
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

As the CI are installed org level, all the CI are enabled on this repo as well
this pr is to help with the CI.
if we don't require any CI, we would have removed them from org level.
if not, this PR is required.

### CODE REVIEW

File: .prow.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+presubmits:
+  - name: pre-commit
+    decorate: true
+    skip_report: false
+    always_run: true
+    context: pre-commit
+    spec:
+      containers:
+        - image: quay.io/thoth-station/thoth-precommit-py38:v0.13.0
```
Comment (codificat) [inline .prow.yaml:9]: The latest version I see [in quay](https://quay.io/repository/thoth-station/thoth-precommit-py38?tab=tags&tag=latest) is:

```suggestion
        - image: quay.io/thoth-station/thoth-precommit-py38:v0.14.3
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - id: check-case-conflict
+      - id: detect-private-key
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: check-yaml
+        args: [--allow-multiple-documents]
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```
Comment (codificat) [inline .pre-commit-config.yaml:24]: The `pre-commit-hooks` repo is duplicate, and a couple of the hooks as well:
```suggestion
      - id: detect-private-key
      - id: check-yaml
        args: [--allow-multiple-documents]
```

**Review by codificat — CHANGES_REQUESTED** [review]
A couple of suggestions

**Comment by tumido** [comment]
@harshad16 can you please fix this PR? It's a great improvement to the CI process and I think it would help to have it merged :slightly_smiling_face:

**Comment by harshad16** [comment]
oh i somehow missed the review.
thanks.

Ready for review @codificat @tumido

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**